src/actions/public/call.ts

```diff
@@ -66,6 +71,27 @@ export type FormattedCall<
   TChain extends Chain | undefined = Chain | undefined,
 > = FormattedTransactionRequest<TChain>
 
+export type AccountStateOverride = {
+  balance?: bigint
+  nonce?: number
+  code?: Hex
+} & (
+  | {
+      /** Fake key-value mapping to override all slots in the account storage before executing the call. */
+      state?: StateMapping
+      stateDiff?: never
+    }
+  | {
+      /** Fake key-value mapping to override individual slots in the account storage before executing the call. */
+      stateDiff?: StateMapping
+      state?: never
+    }
```

Comment:
We can use our `OneOf` type util here.

```suggestion
  OneOf<
    | {
      /** Fake key-value mapping to override all slots in the account storage before executing the call. */
      state?: StateMapping
    }
    | {
      /** Fake key-value mapping to override individual slots in the account storage before executing the call. */
      stateDiff?: StateMapping
    }
  }>
```